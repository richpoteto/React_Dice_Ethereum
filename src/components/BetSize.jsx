import React from 'react';
import PropTypes from 'prop-types';
import ValueSlider from './ValueSlider';

function BetSize({ betSize, updateBetSize }) {
  return (
    <div className="form-group">
      <b>Bet size</b>
      <ValueSlider value={betSize} updateValue={updateBetSize} step={0.1} max={10} addonText="ETH" />
    </div>
  );
}
BetSize.propTypes = {
  betSize: PropTypes.number.isRequired,
  updateBetSize: PropTypes.func.isRequired,
};

export default BetSize;
