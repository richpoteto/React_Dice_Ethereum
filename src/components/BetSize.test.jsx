import React from 'react';
import BetSize from './BetSize';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<BetSize betSize={0.1} updateBetSize={(function(){})} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
