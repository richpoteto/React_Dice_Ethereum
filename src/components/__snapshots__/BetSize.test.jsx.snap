// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <b>
    Bet size
  </b>
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-lg-2"
    >
      <input
        className="form-control"
        onChange={[Function]}
        type="number"
        value={0.1}
      />
    </div>
    <div
      className="col"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rc-slider-rail"
          style={Object {}}
        />
        <div
          className="rc-slider-track"
          style={
            Object {
              "left": "0%",
              "width": "1%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={10}
          aria-valuemin={0}
          aria-valuenow={0.1}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "1%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
    </div>
  </div>
</div>
`;
